vendor/k8s.io/gengo/v2/comments.go

```diff
@@ -81,3 +83,202 @@ func ExtractSingleBoolCommentTag(marker string, key string, defaultVal bool, lin
 	}
 	return false, fmt.Errorf("tag value for %q is not boolean: %q", key, values[0])
 }
+
+// ExtractFunctionStyleCommentTags parses comments for special metadata tags. The
+// marger argument should be unique enough to identify the tags needed, and
+// should not be a marker for tags you don't want, or else the caller takes
+// responsibility for making that distinction.
+//
+// The prefixes argument further selects among comment lines which match the
+// marker.  If this is nil or empty, all lines which match the marker are
+// considered.  If this is specified, only lines with begin with marker +
+// prefix will be considered.  This is useful when a common marker is used which
+// may match lines which fail this syntax (e.g. which predate this definition).
+//
+// This function looks for input lines of the following forms:
+//   - 'marker' + "key=value"
+//   - 'marker' + "key()=value"
+//   - 'marker' + "key(arg)=value"
+//
+// The arg is optional.  If not specified (either as "key=value" or as
+// "key()=value"), the resulting Tag will have an empty Args list.
+//
+// The value is optional.  If not specified, the resulting Tag will have "" as
+// the value.
+//
+// Tag comment-lines may have a trailing end-of-line comment.
+//
+// The map returned here is keyed by the Tag's name without args.
+//
+// A tag can be specified more than one time and all values are returned.  If
+// the resulting map has an entry for a key, the value (a slice) is guaranteed
+// to have at least 1 element.
+//
+// Example: if you pass "+" as the marker, and the following lines are in
+// the comments:
+//
+//	+foo=val1  // foo
+//	+bar
+//	+foo=val2  // also foo
+//	+baz="qux"
+//	+foo(arg)  // still foo
+//
+// Then this function will return:
+//
+//		map[string][]Tag{
+//	 	"foo": []Tag{{
+//				Name: "foo",
+//				Args: nil,
+//				Value: "val1",
+//			}, {
+//				Name: "foo",
+//				Args: nil,
+//				Value: "val2",
+//			}, {
+//				Name: "foo",
+//				Args: []string{"arg"},
+//				Value: "",
+//			}, {
+//				Name: "bar",
+//				Args: nil,
+//				Value: ""
+//			}, {
+//				Name: "baz",
+//				Args: nil,
+//				Value: "\"qux\""
+//		}}
+func ExtractFunctionStyleCommentTags(marker string, prefixes []string, lines []string) (map[string][]Tag, error) {
+	stripComment := func(in string) string {
+		parts := strings.SplitN(in, "//", 2)
```

Comment:
`//` should already be trimmed before passed in?

Reply:
This is for TRAILING comments.  e.,g.

```
// +k8s:optional()   // YOLO
```